File: threedollar-api/src/main/java/com/depromeet/threedollar/api/config/WebConfig.java
package com.depromeet.threedollar.api.config;

import com.depromeet.threedollar.api.config.interceptor.AuthInterceptor;
import com.depromeet.threedollar.api.config.interceptor.UserMetaInterceptor;
import com.depromeet.threedollar.api.config.resolver.GeoCoordinateArgumentResolver;
import com.depromeet.threedollar.api.config.resolver.MapCoordinateArgumentResolver;
import com.depromeet.threedollar.api.config.resolver.UserIdResolver;
import lombok.RequiredArgsConstructor;
import org.springframework.context.MessageSource;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.support.ReloadableResourceBundleMessageSource;
import org.springframework.validation.Validator;
import org.springframework.validation.beanvalidation.LocalValidatorFactoryBean;
import org.springframework.web.method.support.HandlerMethodArgumentResolver;
import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

import java.util.List;

@RequiredArgsConstructor
@Configuration
public class WebConfig implements WebMvcConfigurer {

    private final AuthInterceptor authInterceptor;
    private final UserMetaInterceptor userMetaInterceptor;

    private final UserIdResolver userIdResolver;
    private final GeoCoordinateArgumentResolver geoCoordinateArgumentResolver;
    private final MapCoordinateArgumentResolver mapCoordinateArgumentResolver;

    @Override
    public void addInterceptors(InterceptorRegistry registry) {
        registry.addInterceptor(authInterceptor);
        registry.addInterceptor(userMetaInterceptor);
    }

    @Override
    public void addArgumentResolvers(List<HandlerMethodArgumentResolver> resolvers) {
        resolvers.addAll(List.of(userIdResolver, geoCoordinateArgumentResolver, mapCoordinateArgumentResolver));
    }

    @Override
    public Validator getValidator() {
        LocalValidatorFactoryBean validatorFactoryBean = new LocalValidatorFactoryBean();
        validatorFactoryBean.setValidationMessageSource(validationMessageSource());
        return validatorFactoryBean;
    }

    @Bean
    public MessageSource validationMessageSource() {
        ReloadableResourceBundleMessageSource messageSource = new ReloadableResourceBundleMessageSource();
        messageSource.setBasename("classpath:/messages/validation");
        messageSource.setDefaultEncoding("UTF-8");
        messageSource.setCacheSeconds(10);
        return messageSource;
    }

}

File: threedollar-api-boss/src/main/kotlin/com/depromeet/threedollar/api/boss/config/WebConfig.kt
package com.depromeet.threedollar.api.boss.config

import com.depromeet.threedollar.api.boss.config.interceptor.AuthInterceptor
import com.depromeet.threedollar.api.boss.config.interceptor.UserMetadataInterceptor
import com.depromeet.threedollar.api.boss.config.resolver.BossIdResolver
import com.depromeet.threedollar.api.core.config.resolver.GeoCoordinateArgumentResolver
import com.depromeet.threedollar.api.core.config.resolver.MapCoordinateArgumentResolver
import org.springframework.context.MessageSource
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.support.ReloadableResourceBundleMessageSource
import org.springframework.validation.Validator
import org.springframework.validation.beanvalidation.LocalValidatorFactoryBean
import org.springframework.web.method.support.HandlerMethodArgumentResolver
import org.springframework.web.servlet.config.annotation.InterceptorRegistry
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer

@Configuration
class WebConfig(
    private val authInterceptor: AuthInterceptor,
    private val bossIdResolver: BossIdResolver,
    private val mapCoordinateArgumentResolver: MapCoordinateArgumentResolver,
    private val geoCoordinateArgumentResolver: GeoCoordinateArgumentResolver,
    private val userMetadataInterceptor: UserMetadataInterceptor
) : WebMvcConfigurer {

    override fun addInterceptors(registry: InterceptorRegistry) {
        registry.addInterceptor(authInterceptor)
        registry.addInterceptor(userMetadataInterceptor)
    }

    override fun addArgumentResolvers(resolvers: MutableList<HandlerMethodArgumentResolver>) {
        resolvers.addAll(listOf(bossIdResolver, mapCoordinateArgumentResolver, geoCoordinateArgumentResolver))
    }

    override fun getValidator(): Validator {
        val validatorFactoryBean = LocalValidatorFactoryBean()
        validatorFactoryBean.setValidationMessageSource(validationMessageSource())
        return validatorFactoryBean
    }

    @Bean
    fun validationMessageSource(): MessageSource {
        val messageSource = ReloadableResourceBundleMessageSource()
        messageSource.setBasename("classpath:/messages/validation")
        messageSource.setDefaultEncoding("UTF-8")
        messageSource.setCacheSeconds(60)
        return messageSource
    }

}
